### PR TITLE
Sasha/ip

### PR DESCRIPTION
**Pull request by klizhentas**
* Add IP v4 check
* Switch to dep for dependency management
* Migrate to kubernetes 2.0.0

### CODE REVIEW

File: build.assets/grpc/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+FROM golang:1.7-wheezy
```
Comment (r0mant) [inline build.assets/grpc/Dockerfile]: Since we're migrating to 1.8.3, use it here too?

File: monitoring/ip.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
+package monitoring
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gravitational/satellite/agent/health"
+	pb "github.com/gravitational/satellite/agent/proto/agentpb"
+	"github.com/gravitational/trace"
```
Comment (r0mant) [inline monitoring/ip.go]: Add empty line to separate import grops.

File: monitoring/ip.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+// Check runs a health check and records any errors into the specified reporter.
+func (s *IPForwardChecker) Check(ctx context.Context, reporter health.Reporter) {
+	value, err := Sysctl(IPForwardParam)
+	if err != nil {
+		reporter.Add(NewProbeFromErr(
+			IPForwardCheck,
+			trace.Wrap(
+				trace.ConvertSystemError(err),
```
Comment (a-palchikov) [inline monitoring/ip.go]: Looks like `ConvertSystemError` should accept a format and arguments just like the other trace error construction APIs:
```go
trace.ConvertSystemError(err, "failed to execute sysctl %v", IPForwardParam)
```

File: monitoring/sysctl.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+)
+
+// Sysctl returns kernel parameter by reading proc/sys
+func Sysctl(name string) (string, error) {
+	path := filepath.Clean(filepath.Join("proc", "sys", strings.Replace(name, ".", "/", -1)))
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return "", trace.ConvertSystemError(err)
+	}
+	return string(data[:len(data)-1]), nil
```
Comment (a-palchikov) [inline monitoring/sysctl.go]: You should be careful with empty `data` here

File: monitoring/ip.go (hunk 3 above)
Comment (klizhentas) [inline monitoring/ip.go]: out of scope of this PR

**Pull request merged**